pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -386,17 +401,18 @@ func (s *FlinkStateMachine) updateGenericService(ctx context.Context, app *v1alp
 	return nil
 }
 
-func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1alpha1.FlinkApplication) error {
+func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1alpha1.FlinkApplication) (bool, error) {
 	if s.shouldRollback(ctx, app) {
 		// Something's gone wrong; roll back
-		return s.updateApplicationPhase(ctx, app, v1alpha1.FlinkApplicationRollingBackJob)
+		s.updateApplicationPhase(app, v1alpha1.FlinkApplicationRollingBackJob)
+		return shouldUpdateK8Object, nil
 	}
 
 	// switch the service to point to the new jobmanager
 	hash := flink.HashForApplication(app)
 	err := s.updateGenericService(ctx, app, hash)
 	if err != nil {
-		return err
+		return !shouldUpdateK8Object, err
```

Comment:
Hey `shouldUpdateK8Object` is applicable for different context.

Can you call it `applicationModified`, or `updateApplication`, `shouldUpdateApplication` ?